packages/cli/src/cli.ts

```diff
@@ -124,8 +128,17 @@ ECOM_STORE_ID=${storeId}
         //
       }
     }
-    let serviceAccountJSON : string | null = null;
-    if (argv.gcloud !== false) {
+    let serviceAccountJSON: string | null = null;
+    if (argv.googleapi !== false) {
+      try {
+        const accessToken = await getAccessTokenGCPAndSetIAM(projectId as string, pwd);
+        if (projectId && accessToken) {
+          serviceAccountJSON = await createServiceAccountKeyApi(projectId, accessToken, pwd);
+        }
+      } catch (e) {
+        //
+      }
+    } else if (argv.gcloud !== false) {
```

Comment:
Tem muito código duplicado do `setup-gcp.ts` 🙃 
Na prática os objetos são os mesmos, mas com `gcloud` tem que salvar em um arquivo e executar um comando, via API manda o objeto no body, então o correto não seria manter os métodos já existentes só adicionando um parâmetro opcional `accessToken?: string` ?
Se o token for passado usa-se API, caso contrário CLI ...

Reply:
Nesse trecho também não acho que precisa de checar um argumento novo, talvez nem devesse ter alteração nenhuma aqui

Reply:
No já existente `siginGcloudAndSetIAM` você pode começar testando se `gcloud` existe no terminal (`command -v gcloud`), se existir usamos ele porque ele já gerencia (com segurança) a autenticação do usuário, se não existir você executa o `getGloudAcessToken`, pega o `accessToken` e segue com os métodos nesse caso usando a REST API